From e6e3da4de8fac7f52eb2591cb95a04ab413f8ace Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Olivier=20De=20Canni=C3=A8re?= <olivier.decanniere@qt.io>
Date: Wed, 9 Oct 2024 09:41:27 +0200
Subject: [PATCH] Compiler: Wrap raw string literals in QStringLiteral in
 generated code

Amends e7146cd98700c29802214a39a4cefd9db146c97a

Fixes: QTBUG-129797
Change-Id: I5dc48412cf29bd2de877dd681ea309a6e74d75c5
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
(cherry picked from commit d63c560ec4ce79c1bb24e1a01a20de41706dd039)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmlcompiler/qqmljscodegenerator.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/qmlcompiler/qqmljscodegenerator.cpp b/src/qmlcompiler/qqmljscodegenerator.cpp
index 807168ca09..e5622fe6f9 100644
--- a/src/qmlcompiler/qqmljscodegenerator.cpp
+++ b/src/qmlcompiler/qqmljscodegenerator.cpp
@@ -77,16 +77,16 @@ QString QQmlJSCodeGenerator::compositeListMetaType(const QString &elementName) c
 {
     return u"QQmlPrivate::compositeListMetaType(aotContext->compilationUnit, "_s
             + (m_jsUnitGenerator->hasStringId(elementName)
-                       ? QString::number(m_jsUnitGenerator->getStringId(elementName))
-                       : u'"' + elementName + u'"') + u")"_s;
+                       ? QString::number(m_jsUnitGenerator->getStringId(elementName)) + u')'
+                       : u"QStringLiteral(\"%1\"))"_s.arg(elementName));
 }
 
 QString QQmlJSCodeGenerator::compositeMetaType(const QString &elementName) const
 {
     return u"QQmlPrivate::compositeMetaType(aotContext->compilationUnit, "_s
             + (m_jsUnitGenerator->hasStringId(elementName)
-                       ? QString::number(m_jsUnitGenerator->getStringId(elementName))
-                       : u'"' + elementName + u'"') + u")"_s;
+                       ? QString::number(m_jsUnitGenerator->getStringId(elementName)) + u')'
+                       : u"QStringLiteral(\"%1\"))"_s.arg(elementName));
 }
 
 QString QQmlJSCodeGenerator::metaObject(const QQmlJSScope::ConstPtr &objectType)
-- 
2.47.0

